pkg/tests/ossm/smcp_tls_ssl_test.go

```diff
@@ -4,152 +4,235 @@
-			util.KubeApplyContents("bookinfo", testSSLDeploymentP)
-		} else if env.Getenv("SAMPLEARCH", "x86") == "z" {
-			util.KubeApplyContents("bookinfo", testSSLDeploymentZ)
-		} else {
-			util.KubeApplyContents("bookinfo", testSSLDeployment)
-		}
-		util.CheckPodRunning("bookinfo", "app=testssl")
-
-		log.Log.Info("Check testssl.sh results. Ignore info	Command error")
-		pod, err := util.GetPodName("bookinfo", "app=testssl")
-		util.Inspect(err, "failed to get testssl pod", "", t)
-
-		command := "./testssl/testssl.sh -6 productpage:9080"
-		msg, err := util.PodExec("bookinfo", pod, "testssl", command, false)
-		util.Inspect(err, "failed to execute testssl in pod", "", t)
-		if !strings.Contains(msg, "TLSv1.2") {
-			t.Errorf("Results not include: TLSv1.2")
-			log.Log.Errorf("Results not include: TLSv1.2")
-		}
-		if !strings.Contains(msg, "ECDHE-RSA-AES128-GCM-SHA256") {
-			t.Errorf("Results not include: ECDHE-RSA-AES128-GCM-SHA256")
-			log.Log.Errorf("Results not include: ECDHE-RSA-AES128-GCM-SHA256")
-		}
-		if !strings.Contains(msg, "P-256") {
-			t.Errorf("Results not include: P-256")
-			log.Log.Errorf("Results not include: P-256")
-		}
-	})
-}
+})
+
+func (t test.TestHelper) {
+	NewTest(t).Id("T27").Groups(test.Full, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		n := "bookinfo"
+		t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns)
+				oc.RecreateNamespace(t, ns)
+				oc.ApplyString(t, meshNamespace, smcpName)
+				
+})
+		t.NewSubTest("Testing Operator test for testssl").Run(func(t test.TestHelper) {
+			   t.LogStep("Enable the Service Mesh Control Plane mTLS to true")
+			   retry.UntilSuccess(t, func(t test.TestHelper) {
+			   t.Log("The Service Mesh Control Plane mTLS is true")
+			   oc.MergePatch(t, meshNamespace,
+					   fmt.Sprintf(`smcp/%s`, smcpName),
+					   "merge",
+					   `{"spec":{"security":{"dataPlane":{"mtls":true},"controlPlane":{"mtls":true}}}}'`,
+			   
+			    	) 	
+				})
+		})	
+		})
+
+   		   oc.WaitSMCPReady(t, ns, smcpName)
+			   t.LogStep("Update SMCP spec.security.controlPlane.tls")
+			   oc.MergePatch(t, meshNamespace,
+					   fmt.Sprint(`smcp/%s`, smcpName),
+					   "merge",
+					   `"spec":{"security":{"controlPlane":{"tls":{"maxProtocolVersion":"TLSv1_3"}}}}}'`,
+					   `"minProtocolVersion":"TLSv1_2"`,
+					   `"maxProtocolVersion":"TLSv1_2"`,
+					   `"cipherSuites":["TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]`,
+					   `"ecdhCurves":["CurveP256", "CurveP384"]`,
+						   )
+
+						   oc.MergePatch(t, ns, "smcp/"+smcpName, "merge" )
+						   oc.Log("Deploy bookinfo")
+						   t.LogStep("Install bookinfo with mTLS")
+						   app.InstallAndWaitReady(t, app.BookinfoWithMTLS(ns))
+						   oc.Log("Deploy testssl pod")
+			   oc.ApplyString(t, ns, testSSLDeploymentP)
+			   oc.ApplyString(t, ns, testSSLDeploymentZ)
+			   oc.ApplyString(t, ns, testSSLDeployment)
+				   
+func assertRequestDenied(t test.TestHelper, ns string, curlCommand string) {
```

Comment:
This func is not used in the test, its not needed but instead of this we need to refactor this code, to test using testssl.sh script inside the deployed pod:

```go
util.CheckPodRunning("bookinfo", "app=testssl")

		log.Log.Info("Check testssl.sh results. Ignore info	Command error")
		pod, err := util.GetPodName("bookinfo", "app=testssl")
		util.Inspect(err, "failed to get testssl pod", "", t)

		command := "./testssl/testssl.sh -6 productpage:9080"
		msg, err := util.PodExec("bookinfo", pod, "testssl", command, false)
		util.Inspect(err, "failed to execute testssl in pod", "", t)
		if !strings.Contains(msg, "TLSv1.2") {
			t.Errorf("Results not include: TLSv1.2")
			log.Log.Errorf("Results not include: TLSv1.2")
		}
		if !strings.Contains(msg, "ECDHE-RSA-AES128-GCM-SHA256") {
			t.Errorf("Results not include: ECDHE-RSA-AES128-GCM-SHA256")
			log.Log.Errorf("Results not include: ECDHE-RSA-AES128-GCM-SHA256")
		}
		if !strings.Contains(msg, "P-256") {
			t.Errorf("Results not include: P-256")
			log.Log.Errorf("Results not include: P-256")
		}
```